### PR TITLE
Basic SFTP support, Issue #521

### PR DESCRIPTION
**Pull request by jackschmidt**
This provides basic SFTP support (issue #521 ) using http://github.com/pkg/sftp -- it provides seek and server side file move, but no remote hashes (unsupported in github.com/pkg/sftp, unsupported in openssh server) and no remote directory move (it seemed new, and I had no need of it).

Directory listing uses a wait group, but everything else is just simple sequential wrappers over the sftp client.

### CODE REVIEW

File: sftp/sftp.go [hunk 1]
```diff
@@ -119,7 +119,7 @@ func NewFs(name, root string) (fs.Fs, error) {
 	if sshClient, err := ssh.Dial("tcp", host + ":" + port, config); err != nil {
 		return nil,err
 	} else if sftpClient, err := sftp.NewClient(sshClient); err != nil {
-		sshClient.Close()
+		_ = sshClient.Close()
```
Comment (ncw) [inline sftp/sftp.go]: Don't ignore the error please!

File: sftp/sftp.go [hunk 2]
```diff
@@ -220,25 +220,31 @@ func (f *Fs) List(out fs.ListOpts, dir string) {
 // Put data from <in> into a new remote sftp file object described by <src.Remote()> and <src.ModTime()>
 func (f *Fs) Put(in io.Reader, src fs.ObjectInfo) (fs.Object, error) {
 	debug( f, "Put '" + src.Remote() + "'")
-	f.mkdir( f.sftpClient.Join( f.root, filepath.Dir( src.Remote() ) ) )
+	_ = f.mkdir( f.sftpClient.Join( f.root, filepath.Dir( src.Remote() ) ) )
```
Comment (ncw) [inline sftp/sftp.go]: Don't ignore the error please!

**Comment by ncw** [comment]
I have had a look over the code - looks very nice :-)

I don't think you've generated or run [the unit and integration tests](https://github.com/ncw/rclone/blob/master/CONTRIBUTING.md#writing-a-new-backend)

On that page are some more things that need to be done for making a new Fs too.

I'm happy to do some of those things (eg docs) but can you do the testing parts please? You'll find lots of corner cases in the tests!

If you could squash this into two commits, one for your actual code, and another for all the vendor stuff that will make it easier to review.

Thanks for a super contribution - looking forward to the next version :-)

**Comment by ncw** [comment]
HI!  Just wondering if you are working on the next update of this?

**Comment by nwgat** [comment]
will there be ssh key support? its 2017 after all

**Comment by ncw** [comment]
@jackschmidt I'm going to merge this to a branch and add tests in time for the 1.36 release.  Hopefully that is OK with you!

-- Nick

**Comment by jackschmidt** [comment]
That works. I don't have time to work on it.

**Comment by ncw** [comment]
@jackschmidt 
> That works. I don't have time to work on it.

No problem - thank you for your contribution - much appreciated :-)
